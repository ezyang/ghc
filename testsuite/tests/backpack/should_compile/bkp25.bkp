{-# LANGUAGE NoSignatureMerging #-}
-- This is broken under signature merging

unit p where
    signature A(A) where
        data A
    signature B(A) where
        import A
    module P where
        import A
        import B
        type ZZ = A

unit r where
    module Impl where
        data A = A

unit q where
    include p
    include r
    signature A(A) where
        import Impl(A)
    signature B(A) where
        import Impl(A)
    module M where
        import A
        import B
        import P
        type AA = A
        f :: ZZ -> AA
        f x = x
