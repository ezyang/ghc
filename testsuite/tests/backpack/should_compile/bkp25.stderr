[1 of 3] Processing p
unit p(hole:A, hole:B)
  [1 of 3] Compiling A[sig]           ( p/A.hsig, nothing )
  [2 of 3] Compiling B[sig]           ( p/B.hsig, nothing )
  [3 of 3] Compiling P                ( p/P.hs, nothing )
[2 of 3] Processing r
  Instantiating r
unit r
  [1 of 1] Compiling Impl             ( r/Impl.hs, bkp25-out/r/Impl.o )
[3 of 3] Processing q
unit q(hole:A, hole:B)
- include r [(Impl, r:Impl)]
- include p(hole:A, hole:B) [(P, p(hole:A, hole:B):P)]
  Checking r
  [1 of 3] Compiling A[sig]           ( q/A.hsig, nothing )
  [2 of 3] Compiling B[sig]           ( q/B.hsig, nothing )
  [3 of 3] Compiling M                ( q/M.hs, nothing )
